site/src/Librecores/ProjectRepoBundle/RepoCrawler/RepoCrawler.php

```diff
@@ -13,16 +17,60 @@
  */
 abstract class RepoCrawler
 {
+    /**
+     * @var SourceRepo
+     */
     protected $repo;
+
+    /**
+     * @var MarkupToHtmlConverter
+     */
     protected $markupConverter;
+
+    /**
+     * @var LoggerInterface
+     */
     protected $logger;
 
-    public function __construct(SourceRepo $repo,
-        MarkupToHtmlConverter $markupConverter, LoggerInterface $logger)
+    /**
+     * @var ExecutorInterface
+     */
+    protected $executor;
+
+    /**
+     * @var SourceCrawlerInterface[]
+     */
+    protected $sourceCrawlers;
```

Comment:
Now thats rather confusing. We have a RepoCrawler which itself has SourceCrawlers? How are the tasks between the two separated, and how can this be reflected in the class/interface names?

This class is now (and partially has been) a mix between an actor and an entity representing an repository. Let's continue the discussion on the mailing list where you already came up with a first solution.